src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs

```diff
@@ -1011,6 +1011,17 @@ async Task CloseServiceConnection(IChannelHandlerContext context, Exception caus
             if (!this.ConnectedToService)
             {
                 // closure happened before IoT Hub connection was established or it was initiated due to disconnect
+                if (cause != null)
+                {
+                    string causeScope = "unknown";
+                    if (cause.Data.Contains(OperationScopeExceptionDataKey) && cause.Data[OperationScopeExceptionDataKey] != null)
```

Comment:
IDictionary has a contract of returning null if value is not present, so it should be ok to do smth like
`string causeScope = (string)cause.Data[Key] ?? "unknown";`